### ISSUE TITLE
Missing meson file

### ISSUE THREAD

**Issue by johnny-mac** [rindeal/libfprint-vfs_proprietary-driver#1]
I didn't know this driver existed until today, I have an HP Elitebook 8440p & it has one of these fingerprint readers, I have it enabled on boot-up in the CMOS. I feel like I must be retarded, I can't figure out this meson thing for the life of me. I've taken a look at several PKGBUILDs but nothing works, so I was hoping you could give me a hand if you don't mind, what folder should I be in and what commands should I run? thank you very much for making this thing available by the way.

```
The Meson build system
Version: 0.48.0
Source dir: /home/user/Compiles/libfprint-vfs_proprietary-driver/libfprint
Build dir: /home/user/Compiles/libfprint-vfs_proprietary-driver/build
Build type: native build

ERROR: Missing Meson file in /home/user/Compiles/libfprint-vfs_proprietary-driver/libfprint/meson.build

A full log can be found at /home/user/Compiles/libfprint-vfs_proprietary-driver/build/meson-logs/meson-log.txt

```

**Comment by rindeal** [comment]
> I didn't know this driver existed until today

I've published this driver just now.

> I can't figure out this meson thing for the life of me

The instructions are in the README file of this project and you can see them in action in the scripts used to build it via Travis CI.

**Comment by johnny-mac** [comment]
I don't think I can install the proprietary driver, so maybe this isn't even worth doing... Tried extracting that .rpm but kept failing at it, and I'm on a MUSL libc distro so if it has to have glibc then I'm boned.
```
libfprint/drivers/vfs_proprietary/meson.build:8:1: ERROR:  C library 'vfsFprintWrapper' not found
```

**Comment by johnny-mac** [comment]
```
If that's not the case on your installation you can change the path to the containing directory by defining libvfsFprintWrapper_dir Meson variable prior to calling the subdir Meson command.
```
^ this is what I am having trouble with I suppose... I'm not sure how that command should look, not a fan of meson... how do I define this?

Did you make these drivers yourself? Why is the driver proprietary? I would love to learn more about this stuff. Sorry to trouble you so much, I have a 138a:0007 and thought there'd never be any kind of support.

**Comment by rindeal** [comment]
I've simplified the scripts used to execute all steps to build libfprint with this driver enabled. I've also added verbose comments and put it into a [single file](https://github.com/rindeal/libfprint-vfs_proprietary-driver/blob/master/.ci/script), I can't help you any more than this. Making everything working using those proprietary binaries is far from trivial, so don't expect it to be a 5-minute job, but rest assured that "it works on my machine" using my ebuilds for Portage package manager. On such a machine everything is as simple as issuing:
```sh
$ USE=vfs_proprietary-driver emerge libfprint && systemctl start vcsFPService
```

and running any app which can make use of the fingerprint sensor.

As for the proprietariness, Validity Sensors is one of the many manufacturers which do not provide anything but some proprietary driver which they once stitched together and didn't touch since. Thus unless someone reverse engineer the protocol these sensors communicate with (usually by inspecting raw USB packets, which is obviously very hard and tedious work), putting up with those proprietary binaries is all it remains if you are unwilling to just throw those devices to waste.
